powerranger/view.py

```diff
@@ -0,0 +1,89 @@
+
+_log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Cursor:
+    """Labels for curses' magic values for the cursor setting."""
+    HIDDEN = 0
+    VISIBLE = 1
+    HIGHLY_VISIBLE = 2
+
+
+class View(metaclass=Singleton):
+    """The view from the user's perspective.
+
+    View is a singleton which can be called from anywhere with the only
+    constraint that in its first invocation, it must be passed the
+    _CursesWindow created by curses.initscr()
+    """
+    def __init__(self, stdscr=None):
+        # Intellisense respects Python's privacy conventions, so we use
+        # curses.newwin as a hack which returns curses._CursesWindow for code
+        # completion.
+        self.stdscr: curses.newwin = stdscr
+        self.active_dir = Path(config.DEFAULT_STARTUP_DIR).resolve()
+
+    def render(self):
+        """Render the main window view."""
+        panes = []
+        self.stdscr.clear()
+
+        header_offset = self._draw_header()
+        border_offset = int(config.View.draw_boxes)
+        height = curses.LINES - header_offset
+
+        parent_width = int(curses.COLS * config.View.parent_pane_width_percent)
+        parent_start = 0
+        panes.append(curses.newwin(height, parent_width, header_offset, parent_start))
+
+        # Path.parent will return active_dir if it happens to be the root, so
+        # use parents[0] instead.
+        for index, item in enumerate(Directory(self.active_dir.parents[0])):
+            if index >= height - (2 * border_offset):
+                break
+
+            panes[0].addnstr(index + border_offset, border_offset, item.name, parent_width, item.color)
+
+        active_width = int(curses.COLS * config.View.active_pane_width_percent)
+        active_start = parent_start + parent_width
+        panes.append(curses.newwin(height, active_width, header_offset, active_start))
+
+        for index, item in enumerate(Directory(self.active_dir)):
+            if index >= height - (2 * border_offset):
+                break
+
+            panes[1].addnstr(index + border_offset, border_offset, item.name, active_width, item.color)
+
+        content_width = curses.COLS - (parent_width + active_width)
+        content_start = active_start + active_width
+        panes.append(curses.newwin(height, content_width, header_offset, content_start))
+
+        self.stdscr.refresh()
+
+        for pane in panes:
+            if config.View.draw_boxes:
+                pane.box()
+
+            pane.refresh()
+
+    def _draw_header(self) -> int:
+        """Draw the View's top text and line.
+
+        @return The combined vertical offset
+        """
+        self.stdscr.addnstr(0, 0, str(self.active_dir.absolute()), curses.COLS)
+        self.stdscr.hline(1, 0, curses.A_HORIZONTAL, curses.COLS)
+        return 2
+
+    def resize(self):
+        """Handler for a KEY_RESIZE event."""
+        curses.resize_term(*self.stdscr.getmaxyx())
+        curses.curs_set(Cursor.HIDDEN)
```

Comment:
This doesn't seem to have any effect on Git Bash, but works well on the other two. In any case, we should still position the cursor to follow the focused item later.